utils/misc/misc.go

```diff
@@ -1012,11 +1010,25 @@ func MakeRetryablePostRequest(url, endpoint string, data interface{}) (response
 
 // GetMD5UUID hashes the given string into md5 and returns it as auuid
 func GetMD5UUID(str string) (uuid.UUID, error) {
```

Comment:
Given the changes introduced, it might make sense to write some test to make sure we're meeting the expectations.

Reply:
I have written a fuzzy test to compare this with the previous implementation. Is anything else you have in mind?